src/main/java/com/nutomic/syncthingandroid/fragments/SettingsFragment.java

```diff
@@ -297,6 +318,22 @@ public boolean onPreferenceChange(Preference preference, Object o) {
         return true;
     }
 
+    @Nullable
+    private String formatWifiNameList(Set<String> ssids) {
+        if (ssids != null) {
+            ssids = new TreeSet<>(ssids);
+            StringBuilder sb = new StringBuilder();
+            for (String ssid : ssids) {
+                if (sb.length() > 0) {
+                    sb.append(", ");
+                }
+                sb.append(ssid.replaceFirst("^\"", "").replaceFirst("\"$", ""));
+            }
```

Comment:
To clarify on my earlier comment: I'd handle this by replacing the set entries in place (or into a new set), and then doing `TextUtils.join(", ", ids);` on the result. Bit easier to understand than the length check for commas imo.